staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -296,6 +307,39 @@ func (f *featureGate) Add(features map[Feature]FeatureSpec) error {
 	return nil
 }
 
+func (f *featureGate) OverrideDefault(name Feature, override bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	if f.closed {
+		return fmt.Errorf("cannot override default for feature %q: gates already added to a flag set", name)
+	}
+
+	known := map[Feature]FeatureSpec{}
+	for name, spec := range f.known.Load().(map[Feature]FeatureSpec) {
+		known[name] = spec
+	}
+
+	if spec, ok := known[name]; !ok {
```

Comment:
switch/case for this many different branches

Reply:
I think this is logically

```go
spec, ok := known[name]
if !ok{
 return error
}

switch{
case spec.locked:
  return error
case deprecated:
  message
case GA:
  message
}

set value

```

once nesting is removed

Reply:
OK. As currently written, it's only considered an error if the requested override is different from the locked default. Should that change?

Reply:
> OK. As currently written, it's only considered an error if the requested override is different from the locked default. Should that change?

regarding locked?  Yes.  Behavior in locked is potentially going to get more difficult to reason about as we support "act as version X"

Reply:
Thanks, done.